### ISSUE TITLE
Bug in printing Factor objects when the variables are not str

### ISSUE THREAD

**Issue by ankurankan** [pgmpy/pgmpy#549]
``` python
In [1]: from pgmpy.factors import Factor

In [2]: phi = Factor([1, 2, 3], [2, 2, 2], np.arange(8))

In [3]: print(phi)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-3-78ba4ef7a080> in <module>()
----> 1 print(phi)

/home/ankur/pgmpy/pgmpy/factors/Factor.py in __str__(self)
    672             return self._str(phi_or_p='phi', tablefmt="psql")
    673         else:
--> 674             return self._str(phi_or_p='phi')
    675 
    676     def _str(self, phi_or_p="phi", tablefmt="fancy_grid"):

/home/ankur/pgmpy/pgmpy/factors/Factor.py in _str(self, phi_or_p, tablefmt)
    686         string_header = list(self.scope())
    687         string_header.append('{phi_or_p}({variables})'.format(phi_or_p=phi_or_p,
--> 688                                                               variables=','.join(string_header)))
    689 
    690         value_index = 0

TypeError: sequence item 0: expected str instance, int found
```

**Comment by khalibartan** [comment]
@ankurankan I was about to report this error, I found it when I was working . I'll fix this my PR for docstrings.

**Comment by khalibartan** [comment]
@ankurankan Should I turn variables to a list of string while printing or, change self.variables to list of string when initializing.

**Comment by ankurankan** [comment]
@khalibartan Convert variables into `str` when printing.

**Comment by khalibartan** [comment]
@ankurankan Close this. Fixed in #552

**Comment by ankurankan** [comment]
Fixed
